heartbeat/monitors/maintwin/maintwin.go

```diff
@@ -0,0 +1,103 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package maintwin
+
+import (
+	"time"
+
+	"github.com/teambition/rrule-go"
+)
+
+var weekdayLookup = map[string]rrule.Weekday{
+	"MO": rrule.MO, "TU": rrule.TU, "WE": rrule.WE, "TH": rrule.TH, "FR": rrule.FR, "SA": rrule.SA, "SU": rrule.SU,
+}
+
+type MaintWin struct {
+	Freq       rrule.Frequency `config:"freq" validate:"required"`
+	Dtstart    string          `config:"dtstart" validate:"required"`
+	Interval   int             `config:"interval"`
+	Duration   time.Duration   `config:"duration" validate:"required"`
+	Wkst       rrule.Weekday   `config:"wkst"`
+	Count      int             `config:"count"`
+	Bysetpos   []int           `config:"bysetpos"`
+	Bymonth    []int           `config:"bymonth"`
+	Bymonthday []int           `config:"bymonthday"`
+	Byyearday  []int           `config:"byyearday"`
+	Byweekno   []int           `config:"byweekno"`
+	Byweekday  []string        `config:"byweekday"`
+	Byhour     []int           `config:"byhour"`
+	Byminute   []int           `config:"byminute"`
+	Bysecond   []int           `config:"bysecond"`
+	Byeaster   []int           `config:"byeaster"`
+}
+
+func (mw *MaintWin) Parse() (r *rrule.RRule, err error) {
+
+	dtstart, err := time.Parse(time.RFC3339, mw.Dtstart)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert the string weekdays to rrule.Weekday
+	weekdays := []rrule.Weekday{}
+	for _, wd := range mw.Byweekday {
+		weekdays = append(weekdays, weekdayLookup[wd])
```

Comment:
Can we add a validation here for elements existing on the map?

Reply:
done here [`f581a44` (#41508)](https://github.com/elastic/beats/pull/41508/commits/f581a442d81d86396febd94fad59b63a165b0a46)